server/api/runtime_handlers/base.py

```diff
@@ -968,8 +983,18 @@ def _delete_pod_resources(
                 logger.warning(
                     f"Cleanup failed processing pod {pod.metadata.name}: {repr(exc)}. Continuing"
                 )
+
         # TODO: don't wait for pods to be deleted, client should poll the deletion status
-        self._wait_for_pods_deletion(namespace, deleted_pods, label_selector)
+        timeout = 180
+        force = False
+        # Deleting pods in specific states with non 0 grace period can cause the pods to be stuck in terminating state,
+        # so we're forcing deletion after the grace period passed in this case.
+        if self.pod_grace_period_seconds > 0:
+            timeout = self.pod_grace_period_seconds + 10
+            force = True
+        self._wait_for_pods_deletion(
+            namespace, deleted_pods, label_selector, timeout, force
+        )
```

Comment:
as discussed offline, I think at this point you can force regardless if `self.pod_grace_period_seconds` is set or not.

Reply:
and when you delete the pod few lines above, you can skip using the explicit grace period